### PR TITLE
Performances optimizations Part 1

### PR DESCRIPTION
**Pull request by brg444**
Part 1 of 2. 

The following post aims to highlight development milestones that helped preserve a reliable experience for users of the Bitcoin software client over the years.

### CODE REVIEW

File: _posts/en/2017-03-13-performance-optimizations-1.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+Headers first synchronization severely mitigates this issue using a new method that involves nodes first downloading and validating block headers from a single peer and then fetching block data in parallel from a multitude of others. 
+
+Complaints about initial block download time have been prevalent since the early days of Bitcoin. With headers first synchronization, the software took a major step forward in terms of usability for new users. Rather than wasting many hours on unreliable synchronization, nodes could now leverage their entire network of peers and cut down the bootstrapping time significantly. With the use of smarter algorithms, another asymptotic improvement was made to this crucial aspect of Bitcoin’s long term sustainability. 
+
+##Block file pruning##
+Release: Bitcoin Core 0.11
+
+Block file pruning was first described by Satoshi Nakamoto in his white paper as a potential solution to disk space scarcity. Seven years later, with the blockchain reaching hundreds of gigabytes, the ability to delete raw blocks of old data after validation came as a major boon to users with limited resources.
```
Comment (harding) [inline _posts/en/2017-03-13-performance-optimizations-1.md]: At the time pruning was introduced, the block chain was less than 100 GB and today on my node with txindex it's still "only" 137 GB, which is not "hundreds of gigabytes".

In addition, I feel like this paragraph conflates Nakamoto's proposed pruning (which [doesn't work as designed](http://bitcoin.stackexchange.com/questions/41208/can-a-blockchain-be-created-that-forgets-transactions-whose-outputs-have-been-sp/41224#41224)) with block file pruning which uses the different mechanism described in your subsequent paragraph.

Comment (brg444) [inline _posts/en/2017-03-13-performance-optimizations-1.md]: Added content according to your suggestions and adjusted the paragraph in question. Let me know if that works. Thanks for reviewing!

File: _posts/en/2017-03-13-performance-optimizations-1.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  * [Parallel script verification #2060](https://github.com/bitcoin/bitcoin/pull/2060)
+
+## Headers first synchronization
+Release: Bitcoin Core 0.10
+
+Striving to further improve initial block download time, the Core project introduced in late 2014 an important re-architecture of the mechanism used by nodes to synchronize with the “best chain”. 
+
+Initially, the process of bootstrapping a new Bitcoin client would involve a user fetching block data from a single peer with the consequence that any interruption or decrease in connection quality would significantly stall the process. With an ever-increasing blockchain size, this would result in sometimes massive waiting time for the synchronization to complete, with a large percentage of users reporting up to multi-day periods depending on their hardware.
+
+Headers first synchronization severely mitigates this issue using a new method that involves nodes first downloading and validating block headers from a single peer and then fetching block data in parallel from a multitude of others.
```
Comment (petertodd) [inline _posts/en/2017-03-13-performance-optimizations-1.md]: s/severely mitigates/completely mitigates/

File: _posts/en/2017-03-13-performance-optimizations-1.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+### Further information
+
+ * [Bitcoin-Qt 0.7.0 Release notes](https://bitcoin.org/en/release/v0.7.0#core-bitcoin-handling-and-blockchain-database)
+ * [Fixed vulnerability explanation: Why the signature cache is a DoS protection](https://bitcointalk.org/index.php?topic=136422.0)
+
+## Ultraprune + LevelDB
+Release: Bitcoin-Qt 0.8.0
+
+Ultraprune was one of the first major upgrade to the Bitcoin software aimed at solving the overhead associated with validating transaction data from the blockchain. Previous versions of the Bitcoin reference client would use the entire transaction index which, even back then, resulted in a considerably heavy set of information to work from. One consequence of this was an almost unbearably slow initial block download, which turned out to be a significant hindrance for users looking to bootstrap a new client.
```
Comment (petertodd) [inline _posts/en/2017-03-13-performance-optimizations-1.md]: Let's reword this to say something like "Previous versions of the Bitcoin reference client maintained an index of all transaction outputs, spent or unspent. Ultraprune significantly reduced the size of that index with the insight that you only needed to keep track of unspent outputs, and an output - once spent - can be removed from the indexes entirely."

File: _posts/en/2017-03-13-performance-optimizations-1.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+### Further information
+
+  * [Bitcoin-Qt 0.10.0 Release notes](https://bitcoin.org/en/release/v0.10.0#faster-synchronization)
+  * [Bitcoin.org Developer Guide](https://bitcoin.org/en/developer-guide#headers-first)
+  * [Pieter Wuille's post to the Bitcoin-dev mailing list](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2014-October/006724.html)
+  
+## Block file pruning
+Release: Bitcoin Core 0.11
+
+Pruning of old data was a concept first described by Satoshi Nakamoto in his white paper as a potential solution to disk space scarcity. Unfortunately, the original design was inadequate and could not be implemented as imagined by its creator. Seven years later, with the blockchain reaching more than a hundred gigabyte, the introduction of block file pruning as we know it today presents a major boon to users with limited resources.
```
Comment (petertodd) [inline _posts/en/2017-03-13-performance-optimizations-1.md]: "more than a hundred gigabytes"

File: _posts/en/2017-03-13-performance-optimizations-1.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+Block file pruning leverages previous work with ultraprune; users who have initially downloaded and validated the blockchain may now discard raw data older than 288 blocks. Because those nodes still preserve the full UTXO set, they remain able to validate unspent data, which is enough to provide full guarantees against double-spends.
```
Comment (petertodd) [inline _posts/en/2017-03-13-performance-optimizations-1.md]: I'd word this is "enough to fully validate new blocks" - just mentioning double-spends is understating it.

File: _posts/en/2017-03-13-performance-optimizations-1.md (hunk 3 above)
Comment (CodeShark) [inline _posts/en/2017-03-13-performance-optimizations-1.md]: ACK

File: _posts/en/2017-03-13-performance-optimizations-1.md (hunk 2 above)
Comment (brg444) [inline _posts/en/2017-03-13-performance-optimizations-1.md]: Fixed.

File: _posts/en/2017-03-13-performance-optimizations-1.md (hunk 3 above)
Comment (brg444) [inline _posts/en/2017-03-13-performance-optimizations-1.md]: Thanks. Fixed

File: _posts/en/2017-03-13-performance-optimizations-1.md (hunk 4 above)
Comment (brg444) [inline _posts/en/2017-03-13-performance-optimizations-1.md]: Fixed

File: _posts/en/2017-03-13-performance-optimizations-1.md (hunk 5 above)
Comment (brg444) [inline _posts/en/2017-03-13-performance-optimizations-1.md]: Fixed

File: _posts/en/2017-03-13-performance-optimizations-1.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+As a consequence, synchronization to the tip of the chain happens much faster by making more efficient use of the peer’s resources. During testing of the implementation, developers noted 35% to 100% speed-up when benchmarking against previous versions of the software.
+
+### Further information
+
+  * [Parallel script verification #2060](https://github.com/bitcoin/bitcoin/pull/2060)
+
+## Headers first synchronization
+Release: Bitcoin Core 0.10
+
+Striving to further improve initial block download time, the Core project introduced in late 2014 an important re-architecture of the mechanism used by nodes to synchronize with the “best chain”.
```
Comment (petertodd) [inline _posts/en/2017-03-13-performance-optimizations-1.md]: s/"best chain"/most-work valid chain/

**Pull request merged**